manifests/composer.pp

```diff
@@ -27,7 +27,7 @@
 #   UNIX group of the root user
 #
 class php::composer (
-  String $source                       = $php::params::composer_source,
+  Optional[String[1]] $source          = undef,
```

Comment:
This seems to be a fixed list.  Does it make sense to change it to `Optional[Enum['stable', 'preview', ...]]`?

Reply:
$source contain an URL. may be better to find a type URL (must look on it).
this var may be usefull to override the default selector for a local mirror.

Reply:
D'oh! I mixed up *source* and *channel*, ignore this enum thing.

> $source contain an URL. may be better to find a type URL (must look on it).

[Stdlib::HTTPUrl](https://github.com/puppetlabs/puppetlabs-stdlib#stdlibhttpurl) :wink:

Reply:
This code no more exist. I have transform it in a better form.